src/messaging/message.ts

```diff
@@ -84,19 +84,23 @@ export interface MessageRawPayload extends EntityRawPayload {
     person?: string
   } | null
   readonly statuses?: MessageStatus[]
-  readonly reactions?: Array<{
-    // TODO: parse dates
-    date?: any
-    reaction?: {
-      action?: string
-      type?: string
-      value?: string
-      class?: string
-    } | null
-    external_person_reference_id?: string
-    details?: any
-    payload?: any
-  }>
+  readonly reactions?: MessageReaction[]
+}
+
+export interface MessageReaction {
+  id: string | null
+  // TODO: parse dates
+  date: string | null
+  details: object | null
+  payload: object | null
+  external_person_reference_id: string | null
+  external_person_custom_id: string | null
+  is_staff: boolean
+  reaction: {
+    action: 'react' | 'unreact'
+    type: string | null
+    emoji: string | null
+  } | null
```

Comment:
You're actually right, but in a bit of a different way :) The `MessageRawPayload.reactions` indeed can be `null` if there are no reactions. I just fixed that.

The `MessageReaction.reaction` is `null` by default in the backend. So theoretically it can remain `null`. Practically however, where the reaction is created, the `reaction` field is defined properly. I.e. in incoming webhook parsing, and in react route.

Additionally,  as discussed with @eljefedelrodeodeljefe we're going defensive/pessimistic when dealing with API data. I.e. there is no guarantee. So I think it's useful when IDE complains about trying to access some `messageEntity.reactions[0].reaction.action` without checking whether the properties are not `null`.

WDYT?

Reply:
If I get you right, I would still prefer my proposed version. `readonly reactions?: MessageReaction[] | null` would enable what you want, while still still leaving the option to manually cast/type properly. Lets say you wanted to init a new variable like so `const reaction: MessageReaction = { ...reactionProps }`. That object would potentially be `null` even though it can't be. Correct me if I'm wrong.
We could also never properly cast something `as MessageReaction` when we know it's not null.

It's really a minor difference though, and unlikely to have practical implications. Just my 2 cts :)

Reply:
Yeah I added your proposed version. Thanks!